docs/user_guide/backend_configuration.rst

```diff
@@ -0,0 +1,284 @@
+    )
+
+    time_series = TimeSeries(
+        name="MyTimeSeries",
+        description="A time series from my experiment.",
+        unit="cm/s",
+        data=[1., 2., 3.],
+        timestamps=[0.0, 0.2, 0.4],
+    )
+    nwbfile.add_acquisition(time_series)
+
+    default_backend_configuration = get_default_backend_configuration(
+        nwbfile=nwbfile, backend="hdf5"
+    )
+
+From which a printout of the contents looks like...
+
+.. code-block:: python
+
+    print(default_backend_configuration)
+
+.. code-block:: bash
+
+    Configurable datasets identified using the hdf5 backend
+    -------------------------------------------------------
+
+    acquisition/MyTimeSeries/data
+    -----------------------------
+      dtype : float64
+      full shape of source array : (3,)
+      full size of source array : 24 B
+
+      buffer shape : (3,)
+      expected RAM usage : 24 B
+
+      chunk shape : (3,)
+      disk space usage per chunk : 24 B
+
+      compression method : gzip
+
+    acquisition/MyTimeSeries/timestamps
+    -----------------------------------
+      dtype : float64
+      full shape of source array : (3,)
+      full size of source array : 24 B
+
+      buffer shape : (3,)
+      expected RAM usage : 24 B
+
+      chunk shape : (3,)
+      disk space usage per chunk : 24 B
+
+      compression method : gzip
+
+Customization
+-------------
+
+To modify the chunking or buffering patterns and compression method or options, change those values in the ``.dataset_configurations`` object using the location of each dataset as a specifier.
+
+Let's demonstrate this by modifying everything we can for the ``data`` field of the ``TimeSeries`` object generated above...
+
+.. code-block:: python
+
+    dataset_configurations = default_backend_configuration.dataset_configurations
+    dataset_configuration = dataset_configurations["acquisition/MyTimeSeries/data"]
+    dataset_configuration.chunk_shape = (1,)
+    dataset_configuration.buffer_shape = (2,)
+    dataset_configuration.compression_method = "Zstd"
+    dataset_configuration.compression_options = dict(clevel=3)
+
+Some details to note about what can be changed...
+
+.. note::
+
+    Core fields such as the maximum shape and data type of the source data cannot be altered using this method.
+
+.. note::
+
+    The ``buffer_shape`` must be a multiple of the ``chunk_shape`` along each axis.
+
+.. note::
+
+    You can see what compression methods are available on your installation by examining the following...
+
+    .. code-block:: python
+
+      from neuroconv.tools.nwb_helpers import AVAILABLE_HDF5_COMPRESSION_METHODS
+
+      AVAILABLE_HDF5_COMPRESSION_METHODS
+
+    .. code-block:: bash
+
+      {'gzip': 'gzip',
+       ...
+       'Zstd': hdf5plugin._filters.Zstd}
+
+    And likewise for ``AVAILABLE_ZARR_COMPRESSION_METHODS``.
+
+We can confirm these values are saved by re-printing that particular dataset configuration...
```

Comment:
```suggestion
We can confirm these values are saved by re-printing that particular dataset configuration:
```

It's not clear from this that these configurations made it back into the larger object. It's also not clear how you would use this configuration to modify an in-memory NWBFile object. It would also be great to write this file and then show that the written file followed the chunking and compression parameters we chose. This could be done using HDFView, or some other app or API

Reply:
> It's not clear from this that these configurations made it back into the larger object.

This was obscured by the odd injection of all the notes; with those moved, I put the two printouts side by side and added some spacing

![image](https://github.com/catalystneuro/neuroconv/assets/51133164/9e5655d0-23df-4922-bce4-5618ed6348b6)



> It would also be great to write this file and then show that the written file followed the chunking and compression parameters we chose.

The point the demo here makes is just that they get saved in the `backend_configuration` object itself - I can add something to prove that it actually writes the file in that manner (for chunks and compression anyway, can't prove buffering that way)

Though I will probably just demonstrate it with `h5py` since mentioning/showcasing HDFView could lead to annoyance because if a user wanted to try it they'd need to sign up for HDF5 - also Neurosift doesn't display this information, though I'm sure you could just request that - could also use PyNWB HTML repr after https://github.com/hdmf-dev/hdmf/pull/1100 is merged, but this section is also currently for python script / ipython usage, not specific to a jupyter notebook

Reply:
Added a section to the FAQ for how to check that customizations (or the default for that matter) actually configure the item correctly in the resulting file

Reply:
OK great, could you please add a section on writing the data and confirming (somehow) that the data was written with the proper settings?

Reply:
I did, it's the last section of the FAQ. Uses the file written using the preceeding section

Reply:
OK that section of the FAQ looks great, but I still feel like it would be nice to have another snippet here showing how to apply the configs to an in-memory NWBFile object and write to disk

Reply:
I don't understand what you mean here - what are you imagining and how is it different from the preceding section?